src/Autofac/Features/Decorators/DecoratorMiddleware.cs

```diff
@@ -56,6 +57,14 @@ public DecoratorMiddleware(DecoratorService decoratorService, IComponentRegistra
         /// <inheritdoc/>
         public void Execute(IResolveRequestContext context, Action<IResolveRequestContext> next)
         {
+            if (!(context.Operation is IDependencyTrackingResolveOperation dependencyTrackingResolveOperation))
```

Comment:
I'd suggest moving this check down to after the check for 'is decoration disabled', and keep the style consistent with the other two 'early jump out' checks.

Reply:
Addressed with latest commit